tests/v2/validation/rbac/rbac.go

```
@@ -79,8 +98,77 @@ func createProject(client *rancher.Client, clusterID string) (createProject *man
+	return expectedLabels
+}
+
+func createDeploymentAndWait(steveclient *v1.Client, client *rancher.Client, clusterID string, containerName string, image string, namespaceName string) (*v1.SteveAPIObject, error) {
+	deploymentName := namegen.AppendRandomString("rbac-")
+	containerTemplate := workloads.NewContainer(containerName, image, coreV1.PullAlways, []coreV1.VolumeMount{}, []coreV1.EnvFromSource{})
+	matchLabels := map[string]string{}
+	matchLabels["workload.user.cattle.io/workloadselector"] = fmt.Sprintf("apps.deployment-%v-%v", namespaceName, deploymentName)
+
+	podTemplate := workloads.NewPodTemplate([]coreV1.Container{containerTemplate}, []coreV1.Volume{}, []coreV1.LocalObjectReference{}, matchLabels)
+	deployment := workloads.NewDeploymentTemplate(deploymentName, namespaceName, podTemplate, matchLabels)
+
+	deploymentResp, err := steveclient.SteveType(workloads.DeploymentSteveType).Create(deployment)
+	if err != nil {
+		return nil, err
+	}
+	err = kwait.Poll(5*time.Second, 5*time.Minute, func() (done bool, err error) {
+		deploymentResp, err := steveclient.SteveType(workloads.DeploymentSteveType).ByID(deployment.Namespace + "/" + deployment.Name)
+		if err != nil {
+			return false, nil
+		}
+		deployment := &appv1.Deployment{}
+		err = v1.ConvertToK8sType(deploymentResp.JSONResp, deployment)
+		if err != nil {
+			return false, nil
+		}
+		status := deployment.Status.Conditions
+		for _, statusCondition := range status {
+			if strings.Contains(statusCondition.Message, "forbidden") {
+				err = errors.New(statusCondition.Message)
+				return false, err
+			}
+		}
+		if *deployment.Spec.Replicas == deployment.Status.AvailableReplicas {
+			return true, nil
+		}
+		return false, nil
+	})
+	return deploymentResp, err
+}
+
+func convertNamespace(namespace *v1.SteveAPIObject, steveAdminClient *v1.Client) *coreV1.Namespace {
+	getNSSteveObject, err := steveAdminClient.SteveType(namespaces.NamespaceSteveType).ByID(namespace.ID)
+	if err != nil {
+		return nil
```

I think if there's an error, it should return the error, it is not possible to see the errors.